includes/class-wc-post-data.php

```diff
@@ -30,6 +30,7 @@ public static function init() {
 		add_action( 'shutdown', array( __CLASS__, 'do_deferred_product_sync' ), 10 );
 		add_action( 'set_object_terms', array( __CLASS__, 'force_default_term' ), 10, 5 );
 		add_action( 'set_object_terms', array( __CLASS__, 'delete_product_query_transients' ) );
+		add_action( 'delete_term_relationships', array( __CLASS__, 'delete_product_query_transients' ) );
```

Comment:
this should be hooked to `deleted_term_relationships` instead. can you please change this and update. thanks